### DISCUSSION TITLE
How to add controls to the header cells?

### DISCUSSION THREAD

**Discussion by scarletjack** [glideapps/glide-data-grid#282] [Q&A]
I'd like to add a filter button, among other controls to the header cell. What would be a good way to accomplish this?

Also, is it possible to change the icon for the header menu?

**Comment by jassmith** [comment]
Neither are possible right now without modifying the data grid. I figured someone would want this eventually but being a lazy programmer I decided to wait until it was needed. What do you want to do exactly?

**Reply by scarletjack** [reply]
I currently have a datagrid for which I wrote a filtering system that I'd love to be able to do in Glide's grid, example below.

https://user-images.githubusercontent.com/68049836/165035965-96742bfc-7495-4b0a-a480-007b47652c11.mp4

As well as the sorted icon that I'm unsure about.

**Comment by skelawsky** [comment]
Is it possible to add filtering inside the header menu? [header menu storybook](https://glideapps.github.io/glide-data-grid/?path=/story/glide-data-grid-dataeditor-demos--header-menus)
